runtime/src/main/java/org/corfudb/recovery/RecoveryUtils.java

```diff
@@ -70,7 +70,7 @@ static ILogData getLogData(CorfuRuntime runtime, boolean loadInCache, long addre
         if (loadInCache) {
             return runtime.getAddressSpaceView().read(address);
         } else {
-            return runtime.getAddressSpaceView().fetch(address);
+            return runtime.getAddressSpaceView().fetch(address, false);
```

Comment:
Why does caching determine the behavior of the fetch? 
Why not handle the trimmed exception on the higher level. 
This seems like overloaded semantics to me.

Reply:
We can't handle the trimmed exceptions in the stream layer cause we want to still read the entries (they won't be returned as a TrimmedException was thrown from the address space layer).

Reply:
Actually for the single address fetch we can handle it in upper layers, but for the readRange we can't. I'll modify it on the single read path but don't see an alternative for multi reads.

Reply:
Yes, the batch read semantics can be different,  where only the subset of addresses can be returned and the upper layers can determine if they response batch contains all the addresses in the request.